src/util.ts

```diff
@@ -250,26 +250,30 @@ function querySideloadingRegistry(commands: string[]): Promise<string> {
   });
 }
 
-function addManifestToRegistry(manifestId: string, manifestPath: string): Promise<any> {
-  return querySideloadingRegistry(['Set-ItemProperty -LiteralPath $RegistryPath -Name "' + manifestId + '" -Value "' + manifestPath + '"']);
+async function addManifestToRegistry(manifestId: string, manifestPath: string): Promise<any> {
+  await querySideloadingRegistry(['Remove-ItemProperty -LiteralPath $RegistryPath -Name "' + manifestPath + '" -ErrorAction SilentlyContinue']);
+  return await querySideloadingRegistry(['Set-ItemProperty -LiteralPath $RegistryPath -Name "' + manifestId + '" -Value "' + manifestPath + '"']);
 }
 
 export function getManifestsFromRegistry(): Promise<string[]> {
   return new Promise(async (resolve, reject) => {
     try {
-      const registryOutput = await querySideloadingRegistry(['Get-ItemProperty -LiteralPath $RegistryPath | ConvertTo-Json -Compress']);
+      // NameAndValueOutput would contain some Powershell property, like PSPath, PSDrive.
+      // So we have to call another command to get registry names only.
+      const registryNameOnlyOutput = await querySideloadingRegistry(['Get-Item -Path $RegistryPath | Select-Object -ExpandProperty Property | ConvertTo-Json -Compress']);
+      const registryNameAndValueOutput = await querySideloadingRegistry(['Get-ItemProperty -LiteralPath $RegistryPath | ConvertTo-Json -Compress']);
 
-      if (!registryOutput || registryOutput.indexOf('{') === -1) {
+      if (!registryNameAndValueOutput || registryNameAndValueOutput.indexOf('{') === -1 || !registryNameOnlyOutput) {
         resolve([]);
       }
 
-      const registryJSON = JSON.parse(registryOutput);
+      // if there is a single line of output, the output type is string, otherwise is array of strings.
+      const NameArray = registryNameOnlyOutput.indexOf('[') === -1 ? [JSON.parse(registryNameOnlyOutput)] : JSON.parse(registryNameOnlyOutput);
```

Comment:
The code would be a little more clear if you use a separate variable.
Also, please start the variable names `NameArray` and `NameAndValueDictionary` with a lowercase letter.

```
const nameJson = JSON.parse(registryNameOnlyOutput);
const nameArray = registryNameOnlyOutput.indexOf('[') === -` ? [nameJson] : nameJson;
```